### PR TITLE
Fix Focal deploys broken by Werkzeug 2.0

### PR DESCRIPTION
**Pull request by johnsca**
Yesterday (May 11), Werkzeug 2.0.0 was released which includes a conditional dependency of dataclasses for Python <3.7 but that fails to install on 3.7+. Because the charm snap uses Python 3.6 during the build but Focal includes Python 3.8, deployments there are now broken.

This pins the Werkzeug library and adds tests to validate that the built wheelhouse will successfully install on a given Python version.

Fixes: [lp:1928254][]

[lp:1928254]: https://bugs.launchpad.net/charm-kubernetes-master/+bug/1928254

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -18,26 +18,13 @@ jobs:
...
       run: tox -vve lint
-  unit-test:
-    name: Unit Tests
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python: [3.6, 3.7, 3.8, 3.9]
-    steps:
-      - name: Check out code
-        uses: actions/checkout@v2
```
Comment (chris-sanders) [inline .github/workflows/main.yml:31]: Is this not necessary to get the repository checked out for the tox runs?

**Review by chris-sanders — COMMENTED** [review]
One question, if that's fine then I'm good.

File: .github/workflows/main.yml (hunk 1 above)
Comment (johnsca) [reply .github/workflows/main.yml:31]: I consolidated the lint & unit (and now the wheelhouse) tasks into a single one to speed up the checks on GH (not blocking on separate VMs) and reduce our resource usage.  It's hidden by the upper fold, but that part is still there in the workflow.

Comment (chris-sanders) [reply .github/workflows/main.yml:31]: Oh wait this is just the diff. There is a checkout it's earlier in the code. You combined lint, unit, wheelhouse so the diff just shows the removal b/c the checkout already existed on the lint step.

**Review by chris-sanders — APPROVED** [review]
Looks good

**Comment by johnsca** [comment]
Backported to stable in 0daa2c3

**Pull request merged**